src/main/java/seedu/meal360/Parser.java

```diff
@@ -0,0 +1,641 @@
+                while (recipeList.size() != 0) {
+                    allRecipes += recipeList.deleteRecipe(index).getName() + ", ";
+                }
+                allRecipes = allRecipes.substring(0, allRecipes.length() - 2);
+                return allRecipes;
+            } else {
+                int recipeIndex = 0;
+                for (Recipe recipe : recipeList) {
+                    // find index of recipe we want to delete
+                    if (recipe.getName().equals(recipeToDelete)) {
+                        break;
+                    }
+                    recipeIndex++;
+                }
+                return recipeList.deleteRecipe(recipeIndex).getName();
+            }
+            // user inputted index of recipe in list
+        } else {
+            // deleting a range of recipes
+            if (input[1].length() >= 3) {
+                String rangeRecipes = "";
+                int startIndex = Integer.parseInt(input[1].charAt(0) + "");
+                int endIndex = Integer.parseInt(input[1].charAt(2) + "");
+                startIndex -= 1;
+                endIndex -= 1;
+                int newSize = recipeList.size() - ((endIndex - startIndex) + 1);
+                while (recipeList.size() != newSize) {
+                    rangeRecipes += recipeList.deleteRecipe(startIndex).getName() + ", ";
+                }
+                rangeRecipes = String.valueOf(new StringBuilder(rangeRecipes.substring(0, rangeRecipes.length() - 2)));
+                return rangeRecipes;
+            } else {
+                int recipeIndex = Integer.parseInt(input[1]);
+                // need to subtract 1 since list is 1-based
+                return recipeList.deleteRecipe(recipeIndex - 1).getName();
+            }
+        }
+    }
+
+    public String parseTagRecipe(String[] inputs, RecipeList recipeList) {
+        String tag;
+        boolean isOnlyTagWordInCommand = inputs.length == 1;
+        boolean isAddTag;
+        boolean isRemoveTag;
+
+        if (isOnlyTagWordInCommand) {
+            throw new IllegalArgumentException("Please indicate at least a tag and a recipe.");
+        }
+
+        StringBuilder commandString = new StringBuilder(inputs[1]);
+        for (int i = 2; i < inputs.length; i++) {
+            commandString.append(" ").append(inputs[i]);
+        }
+
+        isAddTag = commandString.indexOf(">>") == -1 && commandString.indexOf("<<") != -1;
+        isRemoveTag = commandString.indexOf("<<") == -1 && commandString.indexOf(">>") != -1;
+
+        if (!(isAddTag || isRemoveTag)) {
+            throw new IllegalArgumentException("Please enter the command in the correct format.");
+        } else if (isAddTag) {
+            tag = parseAddRecipeTag(commandString.toString(), recipeList);
+        } else if (isRemoveTag) {
+            tag = parseRemoveRecipeTag(commandString.toString(), recipeList);
+        } else {
+            throw new IllegalArgumentException("Invalid command.");
```

Comment:
Same issue of magic literals. Try to convert them into constants so that it can help to enhance readability of code